x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/epm/components/package_icon.tsx

```diff
@@ -5,15 +5,22 @@
  */
 import React from 'react';
 import { ICON_TYPES, EuiIcon, EuiIconProps } from '@elastic/eui';
+import { PackageInfo, PackageListItem } from '../../../types';
+import { useLinks } from '../hooks';
 
-const DEFAULT_ICON_TYPE = 'savedObjectsApp';
+type Package = PackageInfo | PackageListItem;
 
 export const PackageIcon: React.FunctionComponent<{
   packageName: string;
-} & Omit<EuiIconProps, 'type'>> = ({ packageName, ...rest }) => {
+  icons?: Package['icons'];
+} & Omit<EuiIconProps, 'type'>> = ({ packageName, icons, ...euiIconProps }) => {
+  const { toImage } = useLinks();
   // try to find a logo in EUI
-  // TODO: first try to find icon in `icons` property
-  const iconType =
-    ICON_TYPES.find(key => key.toLowerCase() === `logo${packageName}`) || DEFAULT_ICON_TYPE;
-  return <EuiIcon type={iconType} {...rest} />;
+  const euiLogoIcon = ICON_TYPES.find(key => key.toLowerCase() === `logo${packageName}`);
+  const svgIcons = icons?.filter(icon => icon.type === 'image/svg+xml');
+  const localIcon = svgIcons && Array.isArray(svgIcons) && svgIcons[0];
+  const pathToLocal = localIcon && toImage(localIcon.src);
+  const euiIconType = pathToLocal || euiLogoIcon || 'savedObjectsApp';
```

Comment:
There are examples, like `postgres` where the registry points to an [image](https://github.com/elastic/package-registry/blob/master/dev/packages/generated/postgresql-1.0.2/img/icon.svg) but EUI also has a logo value (`logoPostgres` on https://elastic.github.io/eui/#/display/icons)

Not sure what the correct precedence is here. Went with preference for value from `icons`.

Reply:
Using euiLogos was convenient in the beginning. We should stop doing this and only rely on the images from the registry.

Reply:
I think we should at least keep it as a fallback if there's no icon path. Better an EUI logo than the default, right?

Reply:
SGTM for now. I'm thinking about making an icon a requirement in the packages. WDYT?